File: treefrog/controllers/jsoncontroller.cpp
#include "jsoncontroller.h"

JsonController::JsonController(const JsonController &)
    : ApplicationController()
{ }

void JsonController::index()
{
    // write codes
}

void JsonController::json()
{
    QVariantMap obj;
    obj["message"] = "Hello, World!";

    setContentType("application/json");
    renderText(jsonEncode(obj), false);
}

// Don't remove below this line
T_REGISTER_CONTROLLER(jsoncontroller)

File: frameworks/PHP/cakephp/src/Controller/JsonController.php
<?php
//
// JSON Encoding Test
//
namespace App\Controller;

use Cake\Controller\Controller;

class JsonController extends Controller {

    public function index() 
    {
        $data = ['message' => 'Hello, World!'];

        return $this->response
            ->withStringBody(\json_encode($data))
            ->withType('application/json');
    }
}
